typescript/sdk/src/deploy/core/AbacusCoreChecker.ts

```diff
@@ -227,4 +228,24 @@ export class AbacusCoreChecker<
       ),
     );
   }
+
+  async checkInterchainGasPaymaster(chain: Chain): Promise<void> {
+    const contracts = this.app.getContracts(chain);
+    if (contracts.interchainGasPaymaster.addresses) {
+      await this.checkUpgradeBeacon(
+        chain,
+        'InterchainGasPaymaster',
+        contracts.interchainGasPaymaster.addresses,
+      );
+    } else {
+      this.violations.push({
+        type: CoreViolationType.InterchainGasPaymasterNotDeployed,
```

Comment:
meh
are we going to create violations like this for all keys?

Reply:
Good point

Reply:
Made it more generic